### PR TITLE
This PR adds Stardist

### LINKED ISSUE #5054: new module: STARDIST
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by migueLib**
## PR checklist

Closes #5054 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/stardist/main.nf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    """
+    stardist-predict2d \\
+        -i $image \\
+        -o . \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        stardist: $VERSION
```
Comment (mashehu) [inline modules/nf-core/stardist/main.nf:33]: please also print the versions of python, tensorflow and tifffile

**Review by mashehu — CHANGES_REQUESTED** [review]

File: modules/nf-core/stardist/main.nf (hunk 1 above)
Comment (migueLib) [reply modules/nf-core/stardist/main.nf:35]: Added

Comment (mashehu) [reply modules/nf-core/stardist/main.nf:35]: nope not like this, but like this: https://nf-co.re/docs/guidelines/components/modules#emission-of-versions

Comment (migueLib) [reply modules/nf-core/stardist/main.nf]: Alright, I fix the emission of versions 😄

**Review by mashehu — APPROVED** [review]

**Pull request merged**